### ISSUE TITLE
shared memory segment creation failure on RHEL 7

### ISSUE THREAD

**Issue by stevekuznetsov** [sclorg/postgresql-container#97]
On a Bitnami RHEL 7 VM with 1CPU, 2GB RAM I see the following when trying to run this container:

```
waiting for server to start....FATAL:  could not create shared memory segment: Invalid argument
DETAIL:  Failed system call was shmget(key=5432001, size=45793280, 03600).
HINT:  This error usually means that PostgreSQL's request for a shared memory segment exceeded your kernel's SHMMAX parameter.  You can either reduce the request size or reconfigure the kernel with larger SHMMAX.  To reduce the request size (currently 45793280 bytes), reduce PostgreSQL's shared memory usage, perhaps by reducing shared_buffers or max_connections.
    If the request size is already small, it's possible that it is less than your kernel's SHMMIN parameter, in which case raising the request size or reconfiguring SHMMIN is called for.
    The PostgreSQL documentation contains more information about shared memory configuration.
pg_ctl: could not start server
Examine the log output.
.... stopped waiting
$ ipcs -lm

------ Shared Memory Limits --------
max number of segments = 4096
max seg size (kbytes) = 4194303
max total shared memory (kbytes) = 1073741824
min seg size (bytes) = 1
```

The request size is smaller than the maximum and larger than the minimum, so I think the `invalid argument` claim for the `shmget` system call may be the real issue. I attempted to run with `--privileged` but it didn't seem that the permissions were the issue, either. Thoughts on whether this is a postgres config issue with the container or not?

**Comment by praiskup** [comment]
On my RHEL7 machine, I see this:

```sh
$ sysctl -a | grep shmmax
kernel.shmmax = 18446744073692774399
```
I'm not sure what Bitnami is, but I would check available memory and shmmax.
This has been fixed by rhbz#1056547.

**Comment by stevekuznetsov** [comment]
`shmmax` is aliased in the output of `ipcs` above, it's ^max seg size (kbytes) = 4194303`
